enterprise/cmd/llm-proxy/internal/auth/auth.go

```diff
@@ -16,58 +16,58 @@ type Authenticator struct {
 	Logger      log.Logger
 	EventLogger events.Logger
 	Sources     actor.Sources
-	Next        http.Handler
 }
 
-var _ http.Handler = &Authenticator{}
-
-func (a *Authenticator) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	token, err := ExtractBearer(r.Header)
-	if err != nil {
-		response.JSONError(a.Logger, w, http.StatusBadRequest, err)
-		return
-	}
-
-	act, err := a.Sources.Get(r.Context(), token)
-	if err != nil {
-		response.JSONError(a.Logger, w, http.StatusUnauthorized, err)
+func (a *Authenticator) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		token, err := ExtractBearer(r.Header)
+		if err != nil {
+			response.JSONError(a.Logger, w, http.StatusBadRequest, err)
+			return
+		}
 
-		err := a.EventLogger.LogEvent(
-			r.Context(),
-			events.Event{
-				Name:       llmproxy.EventNameUnauthorized,
-				Source:     "anonymous",
-				Identifier: "anonymous",
-			},
-		)
+		act, err := a.Sources.Get(r.Context(), token)
 		if err != nil {
-			a.Logger.Error("failed to log event", log.Error(err))
+			response.JSONError(a.Logger, w, http.StatusUnauthorized, err)
+
+			err := a.EventLogger.LogEvent(
+				r.Context(),
+				events.Event{
+					Name:       llmproxy.EventNameUnauthorized,
+					Source:     "anonymous",
+					Identifier: "anonymous",
+				},
+			)
+			if err != nil {
+				a.Logger.Error("failed to log event", log.Error(err))
+			}
+			return
 		}
-		return
-	}
 
-	if !act.AccessEnabled {
-		response.JSONError(
-			a.Logger,
-			w,
-			http.StatusForbidden,
-			errors.New("LLM proxy access not enabled"),
-		)
+		if !act.AccessEnabled {
+			response.JSONError(
+				a.Logger,
```

Comment:
nit - we didn't quite do this before but this might come in handy one day:

```suggestion
				trace.Logger(a.Logger),
```

same with the other callsites here